### PR TITLE
Shortcuts fixes

### PR DESCRIPTION
**Pull request by Marr11317**
fix the scrollbar in the preferences' shortcut list: it was appearing when you clickied several times on the headers.
Remove a few old qDebug()s, change a few foreach`s and simplify things a little.
Add more informations about setting shortcuts in shortcutcapturedialog.ui.

### CODE REVIEW

File: mscore/shortcut.cpp [hunk 1]
```diff
@@ -3879,7 +3879,7 @@ void Shortcut::retranslate()
 
 void Shortcut::refreshIcons()
       {
-      foreach (Shortcut* s, _shortcuts) {
+      for (Shortcut* s: _shortcuts) {
```
Comment (Jojo-Schmitz) [inline mscore/shortcut.cpp]: minor nitpick: we prefer the : to be surrounded by spaces: `for (Shortcut* s : _shortcuts) {`

**Comment by Marr11317** [comment]
@Jojo-Schmitz I added a space.

File: mscore/shortcutcapturedialog.cpp [hunk 2]
```diff
@@ -88,12 +88,10 @@ bool ShortcutCaptureDialog::eventFilter(QObject* /*o*/, QEvent* e)
       {
       if (e->type() == QEvent::KeyPress) {
             QKeyEvent* keyEvent = static_cast<QKeyEvent*>(e);
-            if(keyEvent->key() == Qt::Key_Tab || keyEvent->key() == Qt::Key_Backtab){
-                  QWidget::keyPressEvent(keyEvent);
+            if(!(keyEvent->key() == Qt::Key_Tab || keyEvent->key() == Qt::Key_Backtab)){
```
Comment (anatoly-os) [inline mscore/shortcutcapturedialog.cpp]: You changed the logic here. What was the reason?

File: mscore/shortcutcapturedialog.cpp [hunk 3]
```diff
@@ -116,11 +114,8 @@ void ShortcutCaptureDialog::keyPress(QKeyEvent* e)
 
       k += e->modifiers();
       // remove shift-modifier for keys that don't need it: letters and special keys
-      if ((k & Qt::ShiftModifier) && ((e->key() < 0x41) || (e->key() > 0x5a) || (e->key() >= 0x01000000))) {
-            qDebug() << k;
-      	k -= Qt::ShiftModifier;
-            qDebug() << k;
-            }
+      if ((k & Qt::ShiftModifier) && ((e->key() < 0x41) || (e->key() > 0x5a) || (e->key() >= 0x01000000)))
+            k -= Qt::ShiftModifier;
```
Comment (anatoly-os) [inline mscore/shortcutcapturedialog.cpp]: This debug message is not obsolete and is used in debugging process of the events.

File: mscore/shortcutcapturedialog.cpp [hunk 4]
```diff
@@ -175,21 +170,8 @@ void ShortcutCaptureDialog::keyPress(QKeyEvent* e)
...
-qDebug("capture key 0x%x  modifiers 0x%x virt 0x%x scan 0x%x <%s><%s>",
-      k,
-      int(e->modifiers()),
-      int(e->nativeVirtualKey()),
-      int(e->nativeScanCode()),
-      qPrintable(A),
-      qPrintable(B)
-      );
+
+      nshrtLabel->setText(Shortcut::keySeqToString(key, QKeySequence::NativeText));
```
Comment (anatoly-os) [inline mscore/shortcutcapturedialog.cpp]: Same here.

**Comment by anatoly-os** [comment]
@Marr11317 thank you for the improvements. I left few comments. Could you have a look, please?

File: mscore/shortcutcapturedialog.ui [hunk 5]
```diff
@@ -34,8 +34,12 @@
    </property>
    <item row="0" column="0" colspan="2">
     <widget class="QLabel" name="descrLabel">
+     <property name="toolTip">
+      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;To enter a shortcut, choose a key sequence, for example, Ctrl+A, and press it on your keyboard. &lt;/p&gt;&lt;p&gt;A key sequence can be implemented into another: Ctrl+B, and then Ctrl+C is a valid shortcut.&lt;/p&gt;&lt;p&gt;Up to four combinations can be used: Ctrl+A then Ctrl+B then Ctrl+C then Ctrl+D is also a valid shortcut.&lt;/p&gt;&lt;p&gt;Note: this last shortcut would be noted &amp;quot;Ctrl+A,Ctrl+B,Ctrl+C,Ctrl+D&amp;quot;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Comment (Jojo-Schmitz) [inline mscore/shortcutcapturedialog.ui]: html in translatable taxt is bad. And I don#t think it is really needed here

File: mscore/shortcutcapturedialog.cpp (hunk 2 above)
Comment (Marr11317) [reply mscore/shortcutcapturedialog.cpp]: fewer lines for the same thing.

**Comment by Marr11317** [comment]
The debug things were put back, and the html was simplified to simple text.

**Comment by Marr11317** [comment]
Anything else needed?

**Comment by anatoly-os** [comment]
@Marr11317 damn, I left a comment and didn't start the review, so my comment was in pending state. Sorry, could you please comment on my words or apply changes?

**Comment by anatoly-os** [comment]
Manually merged. See https://github.com/musescore/MuseScore/commit/921d872f8a48cd7948f9c879ed4625cd496e037e

**Pull request closed without merging**